apps/src/gamelab/AnimationPicker/AnimationPreview.jsx

```diff
@@ -0,0 +1,129 @@
+/** @file Render a gallery image/spritesheet as an animated preview */
+
+var MARGIN_PX = 2;
+
+/**
+ * Render an animated preview of a spritesheet at a given size, scaled with
+ * a fixed aspect ratio to fit.
+ */
+var AnimationPreview = React.createClass({
+  propTypes: {
+    width: React.PropTypes.number.isRequired,
+    height: React.PropTypes.number.isRequired,
+    sourceUrl: React.PropTypes.string.isRequired,
+    sourceWidth: React.PropTypes.number.isRequired,
+    sourceHeight: React.PropTypes.number.isRequired,
+    frameWidth: React.PropTypes.number.isRequired,
+    frameHeight: React.PropTypes.number.isRequired,
+    frameCount: React.PropTypes.number.isRequired,
+    frameRate: React.PropTypes.number.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      currentFrame: 0,
+      framesPerRow: 1,
+      scaledSourceWidth: 0,
+      scaledFrameWidth: 0,
+      scaledFrameHeight: 0,
+      extraTopMargin: 0,
+      wrappedSourceUrl: ''
+    };
+  },
+
+  componentWillMount: function () {
+    this.precalculateRenderProps(this.props);
+  },
+
+  componentWillReceiveProps: function (nextProps) {
+    this.precalculateRenderProps(nextProps);
+  },
+
+  componentWillUnmount: function () {
+    if (this.timeout_) {
+      clearTimeout(this.timeout_);
+    }
+  },
+
+  onMouseOver: function () {
+    this.advanceFrame();
+  },
+
+  onMouseOut: function () {
+    this.stopAndResetAnimation();
+  },
+
+  advanceFrame: function () {
+    this.setState({ currentFrame: (this.state.currentFrame + 1) % this.props.frameCount });
+    this.timeout_ = setTimeout(this.advanceFrame, 1000 / this.props.frameRate);
```

Comment:
I think I was moreso wondering if we should just do a `if (this.timeout_) { clearTimeOut(this.timeout_); }` here before setting our new one just in case.